Move domain access fields on node forms

bjaxelsen opened the pull request:
I've made this based on #239. It allows us to configure Domain Access to move the Domain Access fields to an advanced tab on the node forms. This is good for editors who may not understand the meaning of the fields - they can concentrate on the main content fields.

I don't think this interferes in a bad way with the field group module. The advanced groups on the node edit forms is already available and by making this a configurable option it can easily be turned off in case you want to move the fields to some other field group etc.

zerolab commented on domain_access/config/install/domain_access.settings.yml:
```diff
@@ -0,0 +1 @@
+node_advanced_tab: 'TRUE'
```
This needs to be simply `true`.

The tests fail with:

```
Drupal\Core\Config\Schema\SchemaIncompleteException: Schema errors for
    domain_access.settings with the following errors:
    domain_access.settings:node_advanced_tab variable type is string but
    applied schema class is Drupal\Core\TypedData\Plugin\DataType\BooleanData
    in Drupal\Core\Config\Testing\ConfigSchemaChecker-&gt;onConfigSave()
```

zerolab commented:
Checked this on my local test build. Does what it says on the tin. When the setting is enabled, the fields will move to the advanced groups even if they are added to a field group in the form display settings. The only downside is that the field group then appears as empty, with just the title.

I think it would be helpful to add a note on the domain_access settings page about that to avoid any confusion.

bjaxelsen commented:
Yes, I think it makes sense to only move the fields if they are not put in a field group already. I've changed this and modified the help text on the configuration form.

zerolab commented:
I am not entirely sure about this. The core fields that are in the advanced tab do not react to them being in a field group. So maybe this should do the same.

agentrickard commented on domain_access/domain_access.routing.yml:
```diff
@@ -0,0 +1,7 @@
+domain_config.admin:
```
This route is for domain_access, not domain_config.

agentrickard commented on domain/domain.links.task.yml:
```diff
@@ -1,5 +1,6 @@
-entity.domain.edit_form:
```
I don't understand why this task is being removed here.

bjaxelsen replied:
sorry, think I made a mistake here

agentrickard replied:
It doesn't seem to affect anything. See the new PR in https://github.com/agentrickard/domain/pull/264

agentrickard commented:
OK, I've moved this into the general settings branch that I'm working on, Thanks!
